src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Endpoints/Completion.cs

```diff
@@ -271,4 +282,135 @@ private static TextEdit BuildRevertedEdit(TextEdit provisionalTextEdit)
         var formattingOptions = _formattingOptionsProvider.GetOptions(document.TextDocumentIdentifier.Uri);
         return Task.FromResult(formattingOptions);
     }
+
+    [JsonRpcMethod(LanguageServerConstants.RazorSnippetCompletionEndpointName, UseSingleObjectParameterDeserialization = true)]
+    public CompletionItem[]? GetSnippetCompletions(RazorSnippetCompletionParams completionParams, CancellationToken _)
+    {
+        if (completionParams.TextSpan.Length == 0)
+        {
+            return null;
+        }
+
+        var snippets = _snippetCache.GetSnippets();
+
+        if (snippets.IsDefaultOrEmpty)
+        {
+            return null;
+        }
+
+        var langSpecificCompletionsEnum = completionParams.LanguageKind switch
```

Comment:
Nit: The "Enum" suffix here was confusing to me. I'd either expand to "Enumerable", or just remove it.